Add example metadata files

adborden opened the pull request:
These are in yaml just because it's easier to work with. There might be some consolidation with the [2015 metadata](https://github.com/18F/2015-foia/tree/master/contacts/data) that we want to consider.

I made a weak attempt at updating the schemas.md description, but will probably update it based on this PR feedback :)

adborden committed:
Add example metadata files

adborden committed:
Update metadata file description

Include some of changes around describing submission methods. The description
doesn't quite match our examples as we're still figuring out if/how the 2015
scrapper metadata should be incorporated.

adborden committed:
Note on metadata schema origins

Focusing this metadata file on the current needs rather than making them an
iteration on the 2015 work.

adborden committed:
Update metadata schema description

Remove agency contact information for now as it's not relevant for submission.

adborden commented on schemas.md:
```diff
@@ -91,87 +91,54 @@ The FOIA metadata file should instruct the Platform:
-* submission format
-* required form fields
+* agency abbreviation
+* list of departments
```
2015 contact data uses the term "departments" but "components" seems more appropriate.

adborden commented on schemas.md:
```diff
@@ -91,87 +91,54 @@ The FOIA metadata file should instruct the Platform:
-Platform.
+The schema for this metadata file is partially inspired by the [2015 FOIA
+work](https://github.com/18F/2015-foia/tree/master/contacts/data) around
+collecting contact information.
```
The 2015 contact data format is relevant, although I'm not sure to what extent we want to include that information.

adborden committed:
Clarify fields should be included

adborden commented:
There is still an open question of whether this file should exist per agency or per component.

adborden commented on examples/ABMC.yml:
```diff
@@ -0,0 +1,14 @@
+  submission_methods:
+  - submission_format: usps
+    name: Beatriz Collazo
+    title: FOIA Assistant
```
These could just be collapsed into the address lines.

pkarman replied:
You mean including the `name` or `title` as part of `address_lines`? Maybe an example of the alternative would help clarify.

adborden replied:
I was thinking:
```yaml
    address_lines:
    - Beatriz Collazo
    - FOIA Assistant
    - Suite 500
    - 2300 Clarendon Boulevard
    - Arlington, VA 22201
```

Although now realizing the [2015 contact data](https://github.com/18F/2015-foia/blob/master/contacts/data/ABMC.yaml#L4) parses out city/state/zip:

```yaml
address:
    address_lines:
    - Martha R. Sell
    - FOIA Assistant
    - Suite 500
    city: Arlington
    state: VA
    street: 2300 Clarendon Boulevard
    zip: '22201'
```

pkarman replied:
I would find it a little odd if the addressee name was *only* listed under the address key. I like that a person's name gets its own key.

pkarman commented on schemas.md:
```diff
@@ -91,87 +91,56 @@ The FOIA metadata file should instruct the Platform:
 
-#### Email
+A list of methods that the agency supports for receiving a FOIA request. Each
+method represents a different format. Example formats include `usps`, `fax`,
```
I have been using `usps` as shorthand for `paper` so far but I suppose folks might use any other mailing option (FedEx, UPS, hand courier/delivery). So maybe we should say `paper` instead?

adborden replied:
:+1:

pkarman commented on schemas.md:
```diff
@@ -91,87 +91,56 @@ The FOIA metadata file should instruct the Platform:
-* format: `email`
-* email submission address
+If the agency wants to receive requests through the National FOIA Platform, they
+can include a `submission_method` with the `submission_format: platform`.
```
Since the platform isn't a proper submission format itself, but rather a proxy to an actual submission (it receives the submission on behalf of `email` or `web`) maybe the logic needs tweeking here. E.g. a special field `national_platform` boolean `true|false` to indicate whether the platform should attempt to receive and relay the submission or not. Then the platform itself would parse the `submission_method` to figure out how to relay the submission.

pkarman replied:
In addition, if the agency has an `api` format we should probably reserve the option for that?

adborden replied:
> Since the platform isn't a proper submission format itself, but rather a proxy to an actual submission (it receives the submission on behalf of email or web) maybe the logic needs tweeking here. E.g. a special field national_platform boolean true|false to indicate whether the platform should attempt to receive and relay the submission or not. Then the platform itself would parse the submission_method to figure out how to relay the submission.

Right, let me churn on this a bit more.

> In addition, if the agency has an api format we should probably reserve the option for that?

I'm open to that, although do we have any examples of that right now?

pkarman replied:
No existing examples that I know of, though (a) we haven't talked to all the custom solution agencies, and (b) it doesn't hurt to be a little aspirational in specs like these, since sometimes it requires the articulation of an option for others to consider making it real.

pkarman reviewed (commented):
This is a great start. Aside from my comments about `platform` I'd be happy to merge this and use it as the basis for conversations with agencies.

adborden committed:
Refer to post mail as paper

adborden committed:
Platform is not a submission method

Really, considering back to the Portal roots, the Platform is really just
a proxy or pass-through to existing submission methods and systems. By keeping
the details of additional information requested by agencies out of the
submission schema, we can ensure we stick to our goal of using the metadata to
describe the existing process. The inclusion of the additional fields is not so
much to facilitate dynamic forms on the Platform, but more about having agencies
self-describe their existing process which is useful to the FOIA big picture.

adborden committed:
Remove national_portal option

If folks want to change how electronic submissions work, they would alter their
submission methods as part of policy.

pkarman reviewed (approved):
lgtm
